### PR TITLE
Read entire document

### PR DESCRIPTION
**Pull request by tir-bogo**
First file reader to read an entire document

### CODE REVIEW

File: logfile_tests/reader_tests/reader_base_test.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+Module contains tests for ReaderBase
+"""
+
+import pytest
+from logfile.readers.reader_base import ReaderBase
+
+# pylint: disable=redefined-outer-name
+
+
+class MockReaderBase(ReaderBase):
```
Comment (pkhbogo) [inline logfile_tests/reader_tests/reader_base_test.py:11]: Consider using unittest.mock instead.

Comment (tir-bogo) [reply logfile_tests/reader_tests/reader_base_test.py:11]: Did not know that existed, I will look into that for the next pull request

File: logfile_tests/reader_tests/types/read_full_document_test.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    Returns:
+        dict: Files and folders created
+    """
+    main_dir = tmp_path / "main"
+    main_file1 = main_dir / "file1.txt"
+
+    main_dir.mkdir()
+    main_file1.write_text("Beolab90")
+
+    return {
```
Comment (pkhbogo) [inline logfile_tests/reader_tests/types/read_full_document_test.py:25]: Consider using `yield` and clean up temp files afterwards.

File: logfile_tests/reader_tests/reader_base_test.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    Returns:
+        dict: Files and folders created
+    """
+    main_dir = tmp_path / "main"
+    main_file1 = main_dir / "file1.txt"
+
+    main_dir.mkdir()
+    main_file1.write_text("Beolab_50")
+
+    return {
```
Comment (pkhbogo) [inline logfile_tests/reader_tests/reader_base_test.py:35]: Consider using `yield`and clean up temp files afterwards.

**Review by pkhbogo — COMMENTED** [review]
Looks fine.

**Pull request merged**